source/extensions/filters/http/cache/http_cache.h

```diff
@@ -33,18 +33,18 @@ struct LookupResult {
   CacheEntryStatus cache_entry_status_ = CacheEntryStatus::Unusable;
 
   // Headers of the cached response.
-  Http::ResponseHeaderMapPtr headers_;
+  Http::ResponseHeaderMapPtr headers_ = nullptr;
```

Comment:
I think the preferred way in Envoy is to rely on default construction. (Please correct me if I'm mistaken.)

Reply:
That way doesn't compile for me; breaks on -Wmissing-field-initializers.

Reply:
Nothing ever constructed a LookupResult before this, just declared it with everything uninitialized and then set its fields. Is there a problem with doing it this way? We could explicitly define a constructor instead.

Reply:
I don't understand why it doesn't work to just revert all the initializers in this class and let them default-construct. You mentioned that doesn't compile for you which seems strange because it works in main today.

Reply:
What doesn't compile is the new constructor calls to LookupResult that I added in CacheFilterTest. They can't default-construct without these changes because -Wmissing-field-initializers prevents that. They weren't default-constructing before because the existing usage initialized them to indeterminate values.

Reply:
Yeah I bet if you add std::move you have to have a lot more detailed constructors. What's the implication if you drop that?

Other than that, after a quick scan, I just noticed a bunch of new code where you pass LookupResult by value and it might work better to pass by const ref.